core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -177,9 +189,18 @@ private boolean isAliasPresent(Iterable<String> stageOldNameAliasPair) {
   }
 
   Set<String> getInputs() {
-    if (!Strings.isNullOrEmpty(requiredInputs)) {
+    if (!Strings.isNullOrEmpty(requiredInputs) && !containsMacro(JoinerConfig.REQUIRED_INPUTS)) {
       return ImmutableSet.copyOf(Splitter.on(',').trimResults().omitEmptyStrings().split(requiredInputs));
     }
     return ImmutableSet.of();
   }
+
+  void validateJoinKeySchemas(Map<String, Schema> inputSchemas, Map<String, List<String>> joinKeys,
+        FailureCollector collector) {
+      if (!containsMacro(JoinerConfig.JOIN_KEYS) && joinKeys.size() != inputSchemas.size()) {
+        collector.addFailure("There should be join keys present from each stage.",
+            "Ensure join keys are present from each stage.")
+            .withConfigProperty(JoinerConfig.JOIN_KEYS);
+      }
```

Comment:
nit: indentation
Please import the settings jar and format the code.

Reply:
fixed.